lib/index.js

```diff
@@ -107,15 +111,15 @@ exports.register = (server, options, next) => {
                 RbacCore.evaluatePolicy(policy, wrappedDataRetriever, (err, result) => {
 
                     if (err) {
-                        return reply(err);
+                        return options.onError(request, reply, err);
                     }
 
                     if (result === RbacCore.DENY) {
-                        return reply(Boom.create(options.responseCode.onDeny, 'No permissions to access this resource'));
+                        return options.onError(request, reply, Boom.create(options.responseCode.onDeny, 'No permissions to access this resource'));
                     }
 
                     if (result === RbacCore.UNDETERMINED) {
-                        return reply(Boom.create(options.responseCode.onUndetermined, 'Could not evaluate access rights to resource'));
+                        return options.onError(request, reply, Boom.create(options.responseCode.onUndetermined, 'No permissions to access this resource'));
```

Comment:
'Could not evaluate access rights to resource' != 'No permissions to access this resource' ;-)

Please, change the error message back and I will be happy to merge it. Thank you very much for your great contribution.